app/src/main/java/org/jellyfin/mobile/media/car/LibraryBrowser.kt

```diff
@@ -245,143 +235,162 @@ class LibraryBrowser(
-            }
-            includeItemTypes = arrayOf(BaseItemType.MusicAlbum.name)
-            sortBy = arrayOf(ItemSortBy.DatePlayed)
-            sortOrder = SortOrder.Descending
-            recursive = true
-            imageTypeLimit = 1
-            enableImageTypes = arrayOf(ImageType.Primary)
+        val result by itemsApi.getItems(
+            userId = appPreferences.currentUserUuid,
+            parentId = libraryId,
+            artistIds = filterArtist?.let(::listOf),
+            genreIds = filterGenre?.let(::listOf),
+            includeItemTypes = listOf("MusicAlbum"),
+            sortBy = listOf("DatePlayed"),
+            sortOrder = listOf(SortOrder.DESCENDING),
+            recursive = true,
+            imageTypeLimit = 1,
+            enableImageTypes = listOf(ImageType.PRIMARY),
             limit = 100
-        }
-        return apiClient.getItems(query)?.extractItems()?.browsable()
+        )
+
+        return result.extractItems()?.browsable()
     }
 
-    private suspend fun getArtists(libraryId: String): List<MediaBrowserCompat.MediaItem>? {
-        val query = ArtistsQuery().apply {
-            userId = apiClient.currentUserId
-            parentId = libraryId
-            sortBy = arrayOf(ItemSortBy.SortName)
-            sortOrder = SortOrder.Ascending
-            recursive = true
-            imageTypeLimit = 1
-            enableImageTypes = arrayOf(ImageType.Primary)
+    private suspend fun getArtists(libraryId: UUID): List<MediaBrowserCompat.MediaItem>? {
+        val result by artistsApi.getArtists(
+            userId = appPreferences.currentUserUuid,
+            parentId = libraryId,
+            // FIXME missing: sortBy = arrayOf(ItemSortBy.SortName)
+            // FIXME missing: sortOrder = SortOrder.Ascending
+            // FIXME missing: recursive = true
+            imageTypeLimit = 1,
+            enableImageTypes = listOf(ImageType.PRIMARY),
             limit = 100
-        }
-        return apiClient.getArtists(query)?.extractItems(libraryId)?.browsable()
-    }
+        )
 
+        return result.extractItems(libraryId.toString())?.browsable()
+    }
 
-    private suspend fun getGenres(libraryId: String): List<MediaBrowserCompat.MediaItem>? {
-        val query = ItemsByNameQuery().apply {
-            userId = apiClient.currentUserId
-            parentId = libraryId
-            sortBy = arrayOf(ItemSortBy.SortName)
-            sortOrder = SortOrder.Ascending
-            recursive = true
-            imageTypeLimit = 1
-            enableImageTypes = arrayOf(ImageType.Primary)
+    private suspend fun getGenres(libraryId: UUID): List<MediaBrowserCompat.MediaItem>? {
+        val result by genresApi.getGenres(
+            userId = appPreferences.currentUserUuid,
+            parentId = libraryId,
+            // FIXME missing: sortBy = arrayOf(ItemSortBy.SortName)
+            // FIXME missing: sortOrder = SortOrder.Ascending
+            // FIXME missing: recursive = true
```

Comment:
These 3 properties don't exist in the current API, they probably didn't in the old one either so they can probably be removed without any issue.